assets/src/edit-story/app/prepublish/checklistProvider.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../story';
+import Context from './context';
+import prepublishChecklist from '.';
+
+function ChecklistProvider({ children }) {
+  const { story, pages } = useStory(({ state: { story, pages } }) => ({
+    story,
+    pages,
+  }));
+
+  const [currentList, setCurrentList] = useState(
+    prepublishChecklist({ ...story, pages })
```

Comment:
Just thinking out loud here:

Could `prepublishChecklist` perhaps work better as a `usePrePublishChecklist` hook itself?

Otherwise, using a verb for the function name, e.g. `getPrePublishErrors()` might work better.